flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamOperator.java

```diff
@@ -93,6 +93,20 @@
 	//  state snapshots
 	// ------------------------------------------------------------------------
 
+	/**
```

Comment:
This description feels like it states well what it's not intended for, but doesn't clearly describe what it's intended for. 

It would be great if you can add what you wrote in the ticket description here, as of "Some operators maintain some small transient state ...   Rather that persisting that state in a checkpoint, it can make sense to flush the data downstream ...."